sct-commons/src/main/java/org/lfenergy/compas/sct/commons/scl/ldevice/LDeviceAdapter.java

```diff
@@ -81,20 +83,20 @@ public LDeviceAdapter(IEDAdapter parentAdapter, TLDevice currentElem) {
      * Create DataSet and ReportControl Blocks for the HMI with the given FCDAs.
      * DataSet and ReportControl are created in LN0, even if FCDA refers to another LN.
      *
-     * @param fcdas List of FCDA for which we must create the DataSet and ReportControl
+     * @param po object containing list  of FCDA for which we must create the DataSet and ReportControl
      */
-    public void createHmiReportControlBlocks(List<TFCDA> fcdas) {
+    public void createHmiReportControlBlocks(PO po) {
         LN0Adapter ln0 = getLN0Adapter();
         if (!ln0.getDaiModStValValue().map(ActiveStatus::fromValue).map(ActiveStatus.ON::equals).orElse(false)) return;
-        fcdas.stream()
+        po.getFCDAs().getFCDA().stream()
                 .filter(fcda -> getInst().equals(fcda.getLdInst()) && fcda.isSetLnClass())
-                .forEach(fcda -> (fcda.getLnClass().get(0).equals(TLLN0Enum.LLN_0.value()) ?
+                .forEach(fcda -> (fcda.getLnClass().equals(TLLN0Enum.LLN_0.value()) ?
```

Comment:
Is there another name we can use instead of `fcda` since FCDA is a standard object?

Reply:
Then, when `toFCDA(fcdaFilter)` is used in place of `toFCDA(fcda)`  the results are more interesting.
```suggestion
                .forEach(fcdaFilter -> (fcdaFilter.getLnClass().equals(TLLN0Enum.LLN_0.value()) ?
```